From 260bf630800c2e67a7bc6a4bee694a70a40978a8 Mon Sep 17 00:00:00 2001
From: Marie Janssen <jamuraa@google.com>
Date: Wed, 9 Mar 2016 15:31:48 -0800
Subject: [PATCH 04/15] DO NOT MERGE Check size of pin before replying

If a malicious client set a pin that was too long it would overflow
the pin code memory.

Bug: 27411268
Change-Id: I9197ac6fdaa92a4799dacb6364e04671a39450cc
(cherry picked from commit 432da8d61cdcb49a05f91ca9bebcb854a7878591)
---
 btif/src/btif_dm.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/btif/src/btif_dm.c b/btif/src/btif_dm.c
index b2e80d3..498740f 100644
--- a/btif/src/btif_dm.c
+++ b/btif/src/btif_dm.c
@@ -1969,6 +1969,8 @@ bt_status_t btif_dm_pin_reply( const bt_bdaddr_t *bd_addr, uint8_t accept,
                                uint8_t pin_len, bt_pin_code_t *pin_code)
 {
     BTIF_TRACE_EVENT2("%s: accept=%d", __FUNCTION__, accept);
+    if (pin_code == NULL || pin_len > PIN_CODE_LEN)
+        return BT_STATUS_FAIL;
 #if (defined(BLE_INCLUDED) && (BLE_INCLUDED == TRUE))
 
     if (pairing_cb.is_le_only)
-- 
2.7.4

